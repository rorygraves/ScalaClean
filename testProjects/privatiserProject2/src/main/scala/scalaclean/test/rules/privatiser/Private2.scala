package scalaclean.test.rules.privatiser.p2

object Private2 extends App {

  //not used element should be flagged with comments
  def notUsed1 = ???

  val notUsed2 = ???
  var notUsed3 = ???

  Other.used
  Other.usedAndVisible1
  new Other.usedAndVisible2
  new Other.usedAndVisible2a                             {}
  new Other.usedAndVisible3                              {}
  new Other.usedAndVisible3a with Other.usedAndVisible3b {}
}

object Other {

  def used = {
    //content should be unaffected inside a def
    //as the values are not visible
    val x = 1
    var y = 1
    def z = 1

    val a = 1
    var b = 1
    def c = 1

    x + y + z
  }

  object usedAndVisible1 extends AnyRef {
    //content should be affected inside an object
    //as the values are not visible
    val x = 1
    var y = 1
    def z = 1

    val a = 1
    var b = 1
    def c = 1

    x + y + z
  }

  class usedAndVisible2 extends AnyRef {
    //content should be affected inside a class
    //as the values are not visible
    val x = 1
    var y = 1
    def z = 1

    val a = 1
    var b = 1
    def c = 1

    x + y + z
  }

  class usedAndVisible2a extends AnyRef {
    //content should be affected inside a class
    //as the values are not visible
    val x = 1
    var y = 1
    def z = 1

    val a = 1
    var b = 1
    def c = 1

    x + y + z
  }

  trait usedAndVisible3 extends AnyRef {
    //content should be affected inside a trait
    //as the values are not visible
    val x = 1
    var y = 1
    def z = 1

    val a = 1
    var b = 1
    def c = 1

    x + y + z
  }

  trait usedAndVisible3a extends AnyRef {
    //content should be affected inside a trait
    //as the values are not visible
    val Xx = 1
    var Xy = 1
    def Xz = 1

    val Xa = 1
    var Xb = 1
    def Xc = 1

    Xx + Xy + Xz
  }

  trait usedAndVisible3b extends AnyRef {
    //content should be affected inside a trait
    //as the values are not visible
    val Yx = 1
    var Yy = 1
    def Yz = 1

    val Ya = 1
    var Yb = 1
    def Yc = 1

    Yx + Yy + Yz
  }

  //some declarations with internal structure
  //should not be traversed as the content is not visible

  val a = {
    val x = 1
    def y = 1
    var z = 0
    x + y + z
  }

  def b = {
    val x = 1
    def y = 1
    var z = 0
    x + y + z
  }

  var c = {
    val x = 1
    def y = 1
    var z = 0
    x + y + z
  }

  a + b + c
}
