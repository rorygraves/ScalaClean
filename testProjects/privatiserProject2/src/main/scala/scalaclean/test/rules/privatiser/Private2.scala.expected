package scalaclean.test.rules.privatiser.p2

object Private2 extends App {

  //not used element should be flagged with comments
  private[this] def notUsed1 = ???

  private[this] val notUsed2 = ???
  private[this] var notUsed3 = ???

  Other.used
  Other.usedAndVisible1
  new Other.usedAndVisible2
  new Other.usedAndVisible2a                             {}
  new Other.usedAndVisible3                              {}
  new Other.usedAndVisible3a with Other.usedAndVisible3b {}
}

private[p2] object Other {

  private[p2] def used = {
    //content should be unaffected inside a def
    //as the values are not visible
    val x = 1
    var y = 1
    def z = 1

    val a = 1
    var b = 1
    def c = 1

    x + y + z
  }

  private[p2] object usedAndVisible1 extends AnyRef {
    //content should be affected inside an object
    //as the values are not visible
    private val x = 1
    private var y = 1
    private def z = 1

    private[this] val a = 1
    private[this] var b = 1
    private[this] def c = 1

    x + y + z
  }

  private[p2] class usedAndVisible2 extends AnyRef {
    //content should be affected inside a class
    //as the values are not visible
    private val x = 1
    private var y = 1
    private def z = 1

    private[this] val a = 1
    private[this] var b = 1
    private[this] def c = 1

    x + y + z
  }

  protected[Other] class usedAndVisible2a extends AnyRef {
    //content should be affected inside a class
    //as the values are not visible
    private val x = 1
    private var y = 1
    private def z = 1

    private[this] val a = 1
    private[this] var b = 1
    private[this] def c = 1

    x + y + z
  }

  protected[this] trait usedAndVisible3 extends AnyRef {
    //content should be affected inside a trait
    //as the values are not visible
    private val x = 1
    private var y = 1
    private def z = 1

    private[this] val a = 1
    private[this] var b = 1
    private[this] def c = 1

    x + y + z
  }

  protected[this] trait usedAndVisible3a extends AnyRef {
    //content should be affected inside a trait
    //as the values are not visible
    private val Xx = 1
    private var Xy = 1
    private def Xz = 1

    private[this] val Xa = 1
    private[this] var Xb = 1
    private[this] def Xc = 1

    Xx + Xy + Xz
  }

  protected[this] trait usedAndVisible3b extends AnyRef {
    //content should be affected inside a trait
    //as the values are not visible
    private val Yx = 1
    private var Yy = 1
    private def Yz = 1

    private[this] val Ya = 1
    private[this] var Yb = 1
    private[this] def Yc = 1

    Yx + Yy + Yz
  }

  //some declarations with internal structure
  //should not be traversed as the content is not visible

  private val a = {
    val x = 1
    def y = 1
    var z = 0
    x + y + z
  }

  private def b = {
    val x = 1
    def y = 1
    var z = 0
    x + y + z
  }

  private var c = {
    val x = 1
    def y = 1
    var z = 0
    x + y + z
  }

  a + b + c
}
